### DISCUSSION TITLE
Create User through code

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11739] [Q&A]
Folks we try to create user through code like below: 

![image](https://user-images.githubusercontent.com/5202291/169699673-b03b7912-6f24-4ce9-90d1-0ac57d77d539.png)

and assign the existing role of publisher as well 
![image](https://user-images.githubusercontent.com/5202291/169699910-a0cb52f0-d6d0-4cb5-a53e-0ebaeec2c75b.png)

can you share us the sample code to how to achieve:

**Comment by hishamco** [comment]
Go to the users modules, everything is there in the controller

https://github.com/OrchardCMS/OrchardCore/blob/8465fbb4b37a4a52d684c22e0afaeae17cdde8d9/src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs#L335

**Reply by saleemhaq** [reply]
> Go to the users modules, everything is there in the controller
> 
> https://github.com/OrchardCMS/OrchardCore/blob/8465fbb4b37a4a52d684c22e0afaeae17cdde8d9/src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs#L335

And how i assign the existing role as well? on runtime we require to create user and then assign the role as well? please advise.

**Comment by ns8482e** [comment]
There are multiple options to create user programmatically. 

- using C# code - `UserService` to create user and use `UserManager` to assign role
- using workflow - RegisterUser and Assign role task
- using recipe using command  - see https://github.com/OrchardCMS/OrchardCore/discussions/6977#discussioncomment-1852829
